### DISCUSSION TITLE
Use SSH Deploy Key to push

### DISCUSSION THREAD

**Discussion by ghost** [actions/checkout#1270] [Q&A] [answered]
Hi
I'm trying to push local changes using a repository deploy key.

I've added the private key as as repository secret named `PUSH_KEY` and the public key as deploy key with write access to the repository deploy keys.

When running this action:
```
      - name: Checkout
        uses: actions/checkout@v3.3.0
        with:
          ref: test
          ssh-key: "{{ secrets.PUSH_KEY }}"    
      - name: Commit and push
        run: |
          echo "Test" > Test
          git config user.name github-actions
          git config user.email github-actions@github.com
          git add Test
          git commit -m "Test"
          git push
```

The checkout fails with: 

```
Fetching the repository
  /usr/bin/git -c protocol.version=2 fetch --no-tags --prune --progress --no-recurse-submodules --depth=1 origin +refs/heads/last-run*:refs/remotes/origin/last-run* +refs/tags/last-run*:refs/tags/last-run*
  Load key "/home/runner/work/_temp/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx": error in libcrypto
  git@github.com: Permission denied (publickey).
  Error: fatal: Could not read from remote repository.
```

If I've read the documentation correctly this should work (in general) , but I can't figure out what I've done wrong. 
I've already checked the keys multiple times.

My public key looks like:

`ssh-rsa AB3......hGaW=`

The private key looks like:

```
-----BEGIN OPENSSH PRIVATE KEY-----
g4......Fd=
-----END OPENSSH PRIVATE KEY-----
```

**Comment by ghost** [comment, marked as answer]
Got it, remove the quotes:
`ssh-key: {{ secrets.PUSH_KEY }} `

**Reply by matinzd** [reply]
Does this work for pushing to a repository in the next steps too? Or it's only for fetching from the repository?

**Reply by ghost** [reply]
It does, if you checkout using a ssh key it cached and used for all operations including pushing.
